packages/time-select/src/panel.vue

```diff
@@ -0,0 +1,171 @@
+  return formatTime(next)
+}
+export default defineComponent({
+  components: { ElScrollbar },
+  emits: ['pick', 'destroy'],
+  setup(props, ctx) {
+    // data
+    const popperClass = ref('')
+    const start = ref('09:00')
+    const end = ref('18:00')
+    const step = ref('00:30')
+    const value = ref('')
+    const defaultValue = ref('')
+    const visible = ref(false)
+    const minTime = ref('')
+    const maxTime = ref('')
+    const width = ref(0)
+    // computed
+    const items = computed(() => {
+      const start = start.value
+      const end = end.value
+      const step = step.value
+      const result = []
+      if (start && end && step) {
+        let current = start
+        while (compareTime(current, end) <= 0) {
+          result.push({
+            value: current,
+            disabled:
+              compareTime(current, minTime.value || '-1:-1') <= 0 ||
+              compareTime(current, maxTime.value || '100:100') >= 0,
+          })
+          current = nextTime(current, step)
+        }
+      }
+      return result
+    })
+    // methods
+    const handleClick = item => {
+      if (!item.disabled) {
+        ctx.$emit('pick', item.value)
+      }
+    }
+    const handleClear = () => {
+      ctx.$emit('pick', null)
+    }
+    const scrollToOption = (selector = '.selected') => {
+      const menu = ctx.$refs.popper.querySelector('.el-picker-panel__content')
+      scrollIntoView(menu, menu.querySelector(selector))
+    },
+    const handleMenuEnter = () => {
+      const selected =
+        items.value.map(item => item.value).indexOf(this.value) !== -1
+      const hasDefault =
+        items.value.map(item => item.value).indexOf(this.defaultValue) !== -1
+      const option =
+        (selected && '.selected') ||
+        (hasDefault && '.default') ||
+        '.time-select-item:not(.disabled)'
+      ctx.$nextTick(() => scrollToOption(option))
+    }
+    const scrollDown = step => {
+      const items = items.value
+      const length = items.length
+      let total = items.length
+      let index = items.map(item => item.value).indexOf(this.value)
+      while (total--) {
+        index = (index + step + length) % length
+        if (!items[index].disabled) {
+          this.$emit('pick', items[index].value, true)
+          return
+        }
+      }
+    }
+    const isValidValue = date => {
+      return (
+        items.value
+          .filter(item => !item.disabled)
+          .map(item => item.value)
+          .indexOf(date) !== -1
+      )
+    }
+    const handleKeydown = event => {
+      const keyCode = event.keyCode
+      if (keyCode === 38 || keyCode === 40) {
+        const mapping = { 40: 1, 38: -1 }
+        const offset = mapping[keyCode.toString()]
+        scrollDown(offset)
+        event.stopPropagation()
+        return
+      }
+    }
+  },
+  watch: {
+    value(val) {
+      if (!val) return
+      this.$nextTick(() => this.scrollToOption())
+    },
```

Comment:
Since we are using `Composition API` can we use `watch` like 
```suggestion
import { watch, nextTick } from 'vue'
// other parts
setup(props) {
  watch(value, val => {
    if (!val) return
    nextTick(scrollToOption)
  })
}
```
FYI: [Watch API](https://v3.vuejs.org/guide/reactivity-computed-watchers.html#watch)